lcm-logger/lcm_logger.c

```diff
@@ -359,31 +567,49 @@ static void usage()
-            "  -f, --force                Overwrite existing files\n"
-            "  -h, --help                 Shows this help text and exits\n"
+            "  -f, --force                Overwrite existing files.\n"
+            "  -h, --help                 Shows this help text and exits.\n"
             "  -i, --increment            Automatically append a suffix to FILE\n"
             "                             such that the resulting filename does not\n"
             "                             already exist.  This option precludes -f and\n"
-            "                             --rotate\n"
+            "                             --rotate.\n"
+            "\n"
             "  -l, --lcm-url=URL          Log messages on the specified LCM URL\n"
             "  -m, --max-unwritten-mb=SZ  Maximum size of received but unwritten\n"
             "                             messages to store in memory before dropping\n"
             "                             messages.  (default: 100 MB)\n"
+            "\n"
             "      --rotate=NUM           When creating a new log file, rename existing files\n"
-            "                             out of the way and always write to FILE.0.  If\n"
-            "                             FILE.0 already exists, it is renamed to FILE.1.  If\n"
-            "                             FILE.1 exists, it is renamed to FILE.2, etc.  If\n"
-            "                             FILE.NUM exists, then it is deleted.  This option\n"
-            "                             precludes -i.\n"
+            "                             out of the way and always write to FILE.0.\n"
+            "                             If FILE.0 already exists, it is renamed to FILE.1.\n"
+            "                             If FILE.1 exists, it is renamed to FILE.2, etc.\n"
+            "                             If FILE.NUM exists, then it is deleted.\n"
+            "                             This option precludes -i.\n"
+            "\n"
             "      --split-mb=N           Automatically start writing to a new log\n"
             "                             file once the log file exceeds N MB in size\n"
             "                             (can be fractional).  This option requires -i\n"
             "                             or --rotate.\n"
+            "\n"
             "  -q, --quiet                Suppress normal output and only report errors.\n"
             "  -a, --append               Append events to the given log file.\n"
             "  -s, --strftime             Format FILE with strftime.\n"
             "  -v, --invert-channels      Invert channels.  Log everything that CHAN\n"
             "                             does not match.\n"
+            "      --disk-quota=SIZE      Minimum amount of free space the disk written to\n"
+            "                             must maintain. lcm-logger will exit if the quota is\n"
```

Comment:
This sentence is helpful and accurate, but took me a bit to parse. What do you think about rephrasing a bit or adding an example? I'm thinking something like:

>Minimum amount of free space to reserve on the disk being written to. For example, if there is 100GB of free space on the disk at the start of this operation and this option is set to "80GB", then the logger will not write more than 20GB.

I also think it's fine as-is.

Reply:
went with
```
      --disk-quota=SIZE      Minimum amount of free space to reserve on the disk
                             being written to. lcm-logger will exit when it sees
                             the current free disk space has fallen below the
                             quota minimum.
                             For example, given `du -H` reports 15 GB free, and
                             that nothing else is changing the target disk,
                             --disk-quota=10GB means that lcm-logger will not
                             write more than 5 GB before it exits.
```
.